app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java

```diff
@@ -0,0 +1,212 @@
+        // fake data, Edmonton City Centre
+        end.setLocation(53.5438, -113.4923);
+
+        // prompts for fake data verification
+        Toast.makeText(activity, "Location 1: (53.5232, -113.5263)", Toast.LENGTH_SHORT).show();
+        Toast.makeText(activity, "Location 2: (53.5438, -113.4923)", Toast.LENGTH_SHORT).show();
+
+        // TODO create the activity for selecting locations of request (later use case)
+        //    the new activity will handle everything and will return back to use the
+        //    start and end location coordinates
+        // SetLocationsActivity does not exist, placeholder name
+        //   this will be where the user sets the start and end locations of their request
+        // Intent intent = new Intent(MakeRequestActivity.this, SetLocationsActivity.class);
+        // startActivity(intent);
+    }
+
+    /**
+     * When back is pressed
+     *      Pop up an AlertDialog to confirm that the user will cancel their request.
+     *      Return to MainActivity.
+     */
+    public void onBackPressed() {
+        AlertDialog.Builder adb = new AlertDialog.Builder(this);
+        adb.setTitle("Are you sure?");
+        adb.setMessage("Cancel request?");
+        adb.setCancelable(true);
+        final Activity activity = MakeRequestActivity.this;
+        // TODO semantics re: cancel, button text (is it confusing to have "cancel" and "cancel request"?)
+        adb.setPositiveButton("Cancel request", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                activity.finish();
+                Intent intent = new Intent(MakeRequestActivity.this, MainActivity.class);
+                startActivity(intent);
+            }
+        });
+        adb.setNegativeButton("Cancel", null );
+        adb.show();
+    }
+
+    /**
+     * Use the FareCalculator to estimate the fare between the user-selected start and end locations.
+     * The start and end locations must have both been selected before the fare can be estimated.
+     */
+    public void estimateFare() {
+        if(start == null || end == null) {
+            Toast.makeText(activity, "You must first select a start and end location", Toast.LENGTH_SHORT).show();
+        } else {
+            // TODO use FareCalculator once available
+            // the MockFareCalculator generates random numbers so we can see different values on the display
+            MockFareCalculator fc = new MockFareCalculator(start, end);
+            int fareEstimate = fc.getEstimate();
+
+            TextView fareTextView = (TextView) findViewById(R.id.textView_fareEstimate);
+            fareTextView.setText(formatFare(fareEstimate));
+
+            fareEstimated = fareEstimate;
+        }
+    }
+
+    /**
+     * Format the fare we get as a double as a string to be printed on the screen.
+     * This string will be preceded by a dollar sign (are we concerned about locale?)
```

Comment:
Maybe consider locale.
http://stackoverflow.com/questions/7131922/how-to-format-a-float-value-with-the-device-currency-format
There is a built in class called NumberFormat, but I don't see this as a priority.